// @trycourier/courier/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CourierClient = void 0;
var axios_1 = require("axios");
var client_1 = require("./client");
// cannot be `import` as it's not under TS root dir
// tslint:disable-next-line:no-var-requires
var version = require("../package.json").version;
var DEFAULTS = {
    BASE_URL: "https://api.courier.com"
};
exports.CourierClient = function (options) {
    if (options === void 0) { options = {}; }
    var authorizationToken = options.authorizationToken || process.env.COURIER_AUTH_TOKEN;
    if (!authorizationToken) {
        throw new Error("Courier Auth Token is required.");
    }
    var baseURL = options.baseUrl || process.env.COURIER_BASE_URL || DEFAULTS.BASE_URL;
    var axiosInstance = axios_1.default.create({
        baseURL: baseURL,
        headers: {
            Authorization: "Bearer " + authorizationToken,
            "User-Agent": "courier-node/" + version
        }
    });
    var courier = client_1.client({
        httpClient: axiosInstance
    });
    return courier;
};
